Lib/pathlib.py

```diff
@@ -675,22 +581,34 @@ def parts(self):
         # We cache the tuple to avoid building a new one each time .parts
         # is accessed.  XXX is this necessary?
         try:
-            return self._pparts
+            return self._parts_tuple
         except AttributeError:
-            self._pparts = tuple(self._parts)
-            return self._pparts
+            self._parts_tuple = tuple(self._parts)
+            return self._parts_tuple
 
     def joinpath(self, *args):
         """Combine this path with one or several arguments, and return a
         new path representing either a subpath (if all arguments are relative
         paths) or a totally different path (if one of the arguments is
         anchored).
         """
-        return self._make_child(args)
+        drv1, root1, parts1 = self._drv, self._root, self._parts
+        drv2, root2, parts2 = self._parse_args(args)
+        if root2:
+            if not drv2 and drv1:
+                return self._from_parsed_parts(drv1, root2, [drv1 + root2] + parts2[1:])
+        elif drv2:
+            if drv2 == drv1 or self._flavour.normcase(drv2) == self._flavour.normcase(drv1):
+                # Same drive => second path is relative to the first
+                return self._from_parsed_parts(drv1, root1, parts1 + parts2[1:])
+        else:
+            # Second path is non-anchored (common case)
+            return self._from_parsed_parts(drv1, root1, parts1 + parts2)
+        return self._from_parsed_parts(drv2, root2, parts2)
```

Comment:
Having an `else` clause with a `return` statement preceding this makes you do a double-take on what's going on and how you may end up at this line. As such, might as well inline the lines above for logical simplicity rather than avoiding writing the same line twice.

```suggestion
```